department-of-veterans-affairs/vets-api
# API-40450-participant-id-setting-in-bd

## Summary
* Adds extra check to verify there is a participant ID in the bd build body method if the corresponding doc type s being used.
	* some methods pass in a participant D, but some don't and would still want one so this has a fallback to the headers in the case we want one and it was not provided.
	* The job causing this was the v2 poa_form_builder_job which was not passing in a participant ID even though it was using a doc type that would require it.

## Related issue(s)
[API-40450](https://jira.devops.va.gov/browse/API-40450)

## Testing done
* Send in a submit via POA v2
    * verify the participant ID is being set in bd.rb (line 111)

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
	modified:   modules/claims_api/lib/bd/bd.rb

## Acceptance criteria

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
